### PR TITLE
FIX utf-8 characters in the mask mercure does not break counter

### PR DESCRIPTION
**Pull request by ndrosis**

### CODE REVIEW

File: htdocs/core/lib/functions2.lib.php [hunk 1]
```diff
@@ -814,7 +814,7 @@ function get_next_value($db,$mask,$table,$field,$where='',$objsoc='',$date='',$m
     //print "masktri=".$masktri." maskcounter=".$maskcounter." maskraz=".$maskraz." maskoffset=".$maskoffset."<br>\n";
 
     // Define $sqlstring
-    $posnumstart=strrpos($maskwithnocode,$maskcounter);	// Pos of counter in final string (from 0 to ...)
+    $posnumstart=mb_strrpos($maskwithnocode,$maskcounter, 'UTF-8');	// Pos of counter in final string (from 0 to ...)
```
Comment (rdoursenaud) [inline htdocs/core/lib/functions2.lib.php]: mb_\* functions depends on the mbstring extension that may or may not be available.
You should at least check if it's available before using it and fallback to the regular one if unavailable.
(Yeah PHP's unicode support sucks…)

Comment (ndrosis) [inline htdocs/core/lib/functions2.lib.php]: Hi @rdoursenaud when I put Greek characters numbering reaches to 10.
With two ways properly run one is to add the mb_ \* functions and second to activate the php.ini 
mbstring.internal_encoding = UTF-8
mbstring.func_overload = 7

Comment (rdoursenaud) [inline htdocs/core/lib/functions2.lib.php]: Yes, you can fix the issue locally by overloading regular functions with mbstring enabled ones.
But that doesn't change the fact that if this PR is accepted as-is, it's breaking all non mbstring enabled installs by calling an undefined function.
I suggest that kind of code instead:

``` php
 // Use the multibyte enabled function if available or fallback to the regular one
if (function_exists('mb_strrpos')) {
    $posnumstart=mb_strrpos($maskwithnocode,$maskcounter, 'UTF-8');
} else {
    $posnumstart=strrpos($maskwithnocode,$maskcounter);
}
```

Comment (ndrosis) [inline htdocs/core/lib/functions2.lib.php]: Yes I did the test and run without problem.
The code you suggest a better solution.

**Comment by ndrosis** [comment]
@rdoursenaud  thanks for the help.

**Comment by rdoursenaud** [comment]
@ndrosis You're welcome!

**Pull request merged**